PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -615,6 +623,83 @@ bool AliAnalysisTaskEmcalEmbeddingHelper::IsRunInRunlist(const std::string & pat
   return false;
 }
 
+/**
+ * Get the data background run number from the data file path, fRunNumberBgk and 
+ * Determine the MC run block range,  fEmbeddedRunblockMin and  fEmbeddedRunblockMax (used in IsRunInRunblock())
+ * Right now only done once, for the first data file
+ */
+void AliAnalysisTaskEmcalEmbeddingHelper::SetRunblockRange() 
+{
+  Int_t nBlocks = fEmbeddedRunblock.size();
+  
+  if ( nBlocks == 0 ) return ;
+  
+  AliDebug(1,Form("AliAnalysisTaskEmcalEmbeddingHelper::SetRunblockRange() - nRunBlocks %d\n", nBlocks));
+  
+  // Recover the run number from the input data file path.
+  //
+  // InputEvent() seems to be connected to the external event
+  // so one cannot use InputEvent()->GetRunNumber()
+  AliAnalysisManager *mgr = AliAnalysisManager::GetAnalysisManager();
+   
+   if ( !mgr ) return ;
+  
+  Int_t runNumber = mgr->GetRunFromPath(); 
+  if ( fDataRunNumber != runNumber )
+  {
+    //printf("\t Run number, new %d, previous %d\n",runNumber,fDataRunNumber);
+    fDataRunNumber = runNumber;
+  }
+  
+  Int_t  runMin = -1,      runMax = -1;
+  for (Int_t iblock = 0; iblock < nBlocks-1; iblock++) 
+  {
+    std::istringstream(fEmbeddedRunblock.at(iblock  )) >> runMin;
+    std::istringstream(fEmbeddedRunblock.at(iblock+1)) >> runMax;
+    //    runMin = std::stoi(fEmbeddedRunblock.at(iblock  ));
+    //    runMax = std::stoi(fEmbeddedRunblock.at(iblock+1));
+    AliDebug(1,Form("\t block %d, run min %d, run max %d\n",iblock,runMin,runMax));
+    
+    if ( runMin <= fDataRunNumber && runMax > fDataRunNumber ) 
+    {
+      fEmbeddedRunblockMin = runMin;
+      fEmbeddedRunblockMax = runMax;
+      AliDebug(1,Form("\t Selected range %d, run min %d, run max %d\n",
+                      iblock, fEmbeddedRunblockMin, fEmbeddedRunblockMax));
+      
+      return ;
+    }
+  } // block loop
```

Comment:
Also, I'm not sure why you got through all of this work - I thought you were going to go in a different direction.

At this point, you're in a given job (ie only impacting a single job), so when you're in the runblock mode, why not just remove the filenames that you're not interested in from `fFilenames`? Something like `std::remove_if` could do this for you. And then I don't think you need many modifications further on in the code because all filenames are just ones that you want

Reply:
Note this is executed only once in the first event of the job. Here it is the place where the run range block is found based on the data run number. Here, I just to set once the members fEmbeddedRunblockMin and fEmbeddedRunblockMax and fDataRunNumber that are used later. I did like this in case in the future we decide to make this for the case other DataRuns can appear in the analysis.